CONTRIBUTING_CODE.md

```diff
@@ -0,0 +1,188 @@
+# Contributing to Malachite
+
+Thank you for your interest in contributing to Malachite, a Byzantine Fault Tolerant (BFT) consensus engine written in Rust.
+
+This document provides guidelines and instructions to help you set up your development environment and contribute to the project.
+
+## Table of Contents
+
+- [Setup](#setup)
+  - [Prerequisites](#prerequisites)
+  - [Environment Setup](#environment-setup)
+- [Building the Project](#building-the-project)
+- [Running Tests](#running-tests)
+  - [Unit Tests](#unit-tests)
+  - [Integration Tests](#integration-tests)
+  - [Model-Based Tests (MBT)](#model-based-tests-mbt)
+- [Code Style and Guidelines](#code-style-and-guidelines)
+- [Pull Request Process](#pull-request-process)
+- [Continuous Integration](#continuous-integration)
+
+## Setup
+
+### Prerequisites
+
+To build and test Malachite, you need the following tools:
+
+- **Rust**: Install the latest stable Rust toolchain using [rustup](https://rustup.rs/)
+- **Protocol Buffers Compiler (protoc)**: Required for Protobuf message serialization
+- **Node.js**: Required for running [Quint](https://quint-lang.org)
+- **Quint**: A formal specification language used for our model-based tests
+- **cargo-nextest**: An improved test runner for Rust
+
+### Environment Setup
+
+1. **Install Rust**:
+
+   Via [rustup.rs](https://rustup.rs):
+
+   ```bash
+   curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+   ```
+
+2. **Install the Protocol Buffers Compiler**:
+
+   For Ubuntu/Debian:
+
+   ```bash
+   sudo apt-get install protobuf-compiler
+   ```
+
+   For macOS:
+
+   ```bash
+   brew install protobuf
```

Comment:
Might be useful to say which of the tests requires this dependency. Or is it 'cargo build' ?

Reply:
Yeah it's for building the Malachite apps.